core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -531,7 +531,12 @@ public GraphHopper init(GraphHopperConfig ghConfig) {
             throw new IllegalArgumentException("spatial_rules.max_bbox has been deprecated. There is no replacement, all custom areas will be considered.");
 
         setProfiles(ghConfig.getProfiles());
-        flagEncodersString = ghConfig.getString("graph.flag_encoders", flagEncodersString);
+        vehiclesString = ghConfig.getString("graph.vehicles", vehiclesString);
+        if (vehiclesString.isEmpty()) {
+            logger.warn("The option graph.flag_encoders is deprecated and will be removed. Replace with graph.vehicles");
```

Comment:
I assume we want to print the warning when `graph.flag_encoders` is used and not if `graph.vehicles` is **not** used? Actually not using either is what we recommend and in this case of course there should be no warning. And maybe throw an error if **both** are used?